--- a/lib/lzma/LzmaTools.c
+++ b/lib/lzma/LzmaTools.c
@@ -60,7 +60,7 @@ int lzmaBuffToBuffDecompress (unsigned char *outStream, SizeT *uncompressedSize,
     SizeT outSize;
     SizeT outSizeHigh;
     ELzmaStatus state;
-    SizeT compressedSize = (SizeT)(length - LZMA_PROPS_SIZE);
+    SizeT compressedSize = (SizeT)(length - LZMA_DATA_OFFSET);
 
     debug ("LZMA: Image address............... 0x%p\n", inStream);
     debug ("LZMA: Properties address.......... 0x%p\n", inStream + LZMA_PROPERTIES_OFFSET);
